examples/hnas/example_mlp.py

```diff
@@ -0,0 +1,228 @@
+import autogluon.core as ag
+import time
+import logging
+import yaml
+import argparse
+
+import torch
+import torch.nn as nn
+
+from torch.utils.data.sampler import SubsetRandomSampler
+
+from torchvision import datasets
+from torchvision import transforms
+
+
+@ag.args(
+    n_units_1=ag.space.Int(lower=4, upper=1024),
+    n_units_2=ag.space.Int(lower=4, upper=1024),
+    dropout_1=ag.space.Real(lower=0, upper=.99),
+    dropout_2=ag.space.Real(lower=0, upper=.99),
+    learning_rate=ag.space.Real(lower=1e-6, upper=1, log=True),
+    batch_size=ag.space.Int(lower=8, upper=128),
+    wd=ag.space.Real(lower=1e-8, upper=1, log=True),
+    epochs=27
+)
+def objective_function(args, reporter, **kwargs):
+    ts_start = time.time()
+
+    # Hyperparameters to be optimized
+    n_units_1 = args.n_units_1
+    n_units_2 = args.n_units_2
+    dropout_1 = args.dropout_1
+    dropout_2 = args.dropout_2
+    batch_size = args.batch_size
+    learning_rate = args.learning_rate
+    wd = args.wd
+
+    # Make sure you downloaded the dataset before running AutoGluon. Downloading the data inside the
```

Comment:
Can you give a more precise instruction of what the user has to do?

Reply:
ok I added some python commands to load this data

Reply:
You could add these commands to download MNIST to the main function, this would then make sure the example runs on a single instance. The tricky part is if somebody runs this on multiple instances.